common/src/buffers/in_mem/send.rs

```diff
@@ -0,0 +1,233 @@
+use crate::buffers::{DeniablePayload, DenimChunk, Flag, SendingBuffer};
+use crate::denim_message::DeniableMessage;
+use crate::error::LibError;
+use async_trait::async_trait;
+use bincode::config;
+use prost::Message;
+use rand::RngCore;
+use std::collections::VecDeque;
+use std::mem::take;
+
+const DENIM_CHUNK_WITHOUT_PAYLOAD: usize = 4;
+
+const DENIABLE_PAYLOAD_MIN_LENGTH: usize = 20;
+
+struct Buffer {
+    content: Vec<u8>,
+    message_id: u32,
+    next_sequence_number: u8,
+}
+
+pub struct InMemorySendingBuffer {
+    q: f32,
+    outgoing_messages: VecDeque<DeniableMessage>,
+    buffer: Buffer,
+}
+
+#[async_trait]
+impl SendingBuffer for InMemorySendingBuffer {
+    async fn get_deniable_payload(
+        &mut self,
+        reg_message_len: u32,
+    ) -> Result<Option<DeniablePayload>, LibError> {
+        if self.q == 0.0 {
+            return Ok(None);
+        }
+
+        let mut available_bytes = self.calculate_deniable_payload_length(reg_message_len);
+
+        if available_bytes < DENIABLE_PAYLOAD_MIN_LENGTH {
+            return Ok(Some(
+                DeniablePayload::builder()
+                    .denim_chunks(vec![])
+                    .garbage(InMemorySendingBuffer::create_n_random_bytes(
+                        available_bytes,
+                    ))
+                    .build(),
+            ));
+        }
+
+        let mut denim_chunks: Vec<Vec<u8>> = Vec::new();
+
+        while available_bytes > DENIM_CHUNK_WITHOUT_PAYLOAD {
+            let deniable_payload_len = available_bytes - DENIM_CHUNK_WITHOUT_PAYLOAD;
+
+            let chunk = self.get_next_chunk(deniable_payload_len);
+
+            match chunk {
+                None => {
+                    break;
+                }
+                Some(chunk) => {
+                    let encoded_chunk = bincode::encode_to_vec(chunk, config::standard())
```

Comment:
please make this to a function on the DenimChunk so it consumes self and returns a bytearray